web/profiles/custom/yalesites_profile/modules/custom/ys_themes/ys_themes.module

```diff
@@ -138,6 +138,26 @@ function ys_themes_preprocess_block(&$variables) {
   }
 }
 
+/**
+ * Implements hook_preprocess_node().
+ */
+function ys_themes_preprocess_node(&$variables) {
+  // Add the cache tag, so that the theme setting information is rebuilt when
+  // the config is saved.
+  // Via: https://drupal.stackexchange.com/questions/266379/how-to-clear-cache-for-config-entity-after-making-changes
+  $config = \Drupal::config('ys_themes.theme_settings');
+  \Drupal::service('renderer')->addCacheableDependency($variables, $config);
```

Comment:
Great technique. I wonder if this should be moved to the preprocess_page? Right now this is only clearing the cache for nodes. So caches or will clear for other entities or for dynamically generated pages like the 404 for the search results pages.

Reply:
Ahh, great idea - moved into the `preprocess_page`